edgelet/iotedged/src/lib.rs

```diff
@@ -209,10 +247,21 @@ impl Main {
             }
         }
 
+        let proxy_uri = env::var("HTTPS_PROXY")
+            .or_else(|_| env::var("https_proxy"))
+            .ok();
+        let proxy_uri = match proxy_uri {
+            None => None,
+            Some(s) => {
+                let proxy = s.parse::<Uri>()?;
+                info!("Detected HTTPS proxy server {}", proxy.to_string());
+                Some(proxy)
+            },
+        };
```

Comment:
As it always is in Rust, you can make this more terse:

        let proxy_uri = match env::var("HTTPS_PROXY")
            .or_else(|_| env::var("https_proxy"))
            .ok() {
            None => None,
            Some(s) => {
                let proxy = s.parse::<Uri>()?;
                info!("Detected HTTPS proxy server {}", proxy.to_string());
                Some(proxy)
            },
        };

I'll bet someone else can make it moreso.

Reply:
I had it more terse than that, but the pesky `?` operator messed it up. Would love to know how to work around that...

Reply:
Can you use map here?

Reply:
The idiomatic way might be something like:
```rust
let proxy_uri = env::var("HTTPS_PROXY")
  .or_else(|_| env::var("https_proxy"))
  .and_then(|uri| uri.parse::<Uri>())
  .ok();
```
You might also need to add some `map_err`s in order to get the error types squared away.

Reply:
It gets a little complicated because errors returned from `env::var` need to be converted to `Option::None`, but an error from `parse::<Uri>()` needs to be propagated. I'll keep it the way it is for now, but would love to find another way in the future...